snapshot/src/main/scala/com/evolutiongaming/kafka/journal/SnapshotStoreFlat.scala

```diff
@@ -0,0 +1,58 @@
+package com.evolutiongaming.kafka.journal
+
+import com.evolutiongaming.kafka.journal.eventual.EventualPayloadAndType
+
+/** Snapshot storage for Kafka Journal.
+  *
+  * Uses `Key` instead of `persistenceId`, similar to [[ReplicatedJournalFlat]]
+  * and [[EventualJournal]].
+  */
+trait SnapshotStoreFlat[F[_]] {
```

Comment:
Am I right that later you're planning to introduce non-flat trait?

I'm asking cos earlier you've added scaladocs where explicitly mentioned:
> The flat interface was replaced by hierarchical [[ReplicatedJournal]] for
> all means and purposes except for the unit tests, and should not be used
> directly anymore.

Do we need flat trait at all?

I've seen reasoning in PR description:
> I also tried to make it look similar to ReplicatedJournalFlat and EventualJournal for sake of uniformity of the codebase.

and would vote for keeping the implementation simple. Might be better would be dropping other flat traits then introducing new one? BUT **I don't insist**, feel free to resolve the comment 👍

Reply:
I was thinking about having the flat one only.

The hierarchical structure of `ReplicatedJournal` kind of makes sense, because it is writing to Kafka and therefore has to deal with topics and partitions.

`SnapshotStoreFlat` implementation uses Cassandra only and just uses `(topic, id)` tuple as a key, so hierarchy won't add much except a bit more convoluted interface. That why, I suspect, @t3hnar never made `EventualJournal` hierarchical, because it also deals with Cassandra only.

I could rename it to `SnapshotStore` therefore, though I am a bit weary that there is `akka.persistence.snapshot.SnapshotStore` already and if this is going to be too confusing. May be some other name could be more appropriate.

Reply:
minor: I saw others are naming their impls as `SnapshotStore`, package name should make it distinguishable :)

Reply:
IMO: I struggled yesterday with 3 classes from different libs (and packages) all named as `EventSourcedOf` :)

Reply:
@rtar, I'm fine with `SnapshotStore`, hope the only place where one would case of distinguish them is akka-effect/persistence module